### ISSUE TITLE
S2068 FP: should not raise for empty string

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#2387]
### Description

Rule https://rules.sonarsource.com/csharp/RSPEC-2068

If a `password` field is initialized with `null` or an empty string, it should not be considered as a hardcoded password.

### Repro steps

Add the below line in sonar-dotnet\sonaranalyzer-dotnet\tests\SonarAnalyzer.UnitTest\TestCases\DoNotHardcodeCredentials_DefaultValues.cs

```
            string password2 = "";
```

### Expected behavior

Should not raise.

### Actual behavior

Raises `'Remove hard-coded password(s): 'password'.'`
Also, it should raise `password2` rather than `password` (the variable name, not the substring name we search with).

### Known workarounds

None

### Related information

* SonarC# Version: 7.13

**Comment by christophe-zurn-sonarsource** [comment]
Regarding the message part, if we harmonize with other analyzer, it will be like this:
`'password' detected in this expression, review this potentially hardcoded credential.`
The word mentioned will still be the subpart of the variable/string literal, but the message will be clearer
